src/collect.nim

```diff
@@ -41,23 +41,28 @@ proc canWrite(plugin: Plugin, key: string, decls: seq[string]): bool =
   if key notin decls and "*" notin decls:
     error("Plugin '" & plugin.name & "' produced undeclared key: '" & key & "'")
     return false
-  if not spec.system:
+  if not get[bool](spec, "system"):
     return true
 
   case plugin.name
   of "system", "metsys":
     return true
   of "conffile":
-    if spec.confAsSystem:
+    if get[bool](spec, "conf_as_system"):
       return true
   else: discard
 
   error("Plugin '" & plugin.name & "' can't write system key: '" & key & "'")
   return false
 
-proc registerKeys(templ: MarkTemplate | ReportTemplate) =
-  for name, content in templ.keys:
-    if content.use: subscribedKeys[name] = true
+proc registerKeys(templ: AttrScope) =
+  let keyOpt = getObjectOpt(templ, "key")
+  if keyOpt.isSome():
+    let key = keyOpt.get()
+    for name, content in key.contents:
+      if content.isA(AttrScope):
+        let useOpt = getOpt[bool](content.get(AttrScope), "use")
+        if useOpt.isSome() and useOpt.get(): subscribedKeys[name] = true
```

Comment:
```suggestion
        if useOpt.isSome() and useOpt.get():
          subscribedKeys[name] = true
```

previously it was an easy to read condition as there was only one `if <condition>:` whereas now its multiple conditions which makes it harder to read whats actually getting done

p.s. in general Im not a fan of single-line conditions and prefer to indent the code block to make it visually different